src/CRVBadDebtRepayment.sol

```diff
@@ -91,15 +94,17 @@ contract CRVBadDebtRepayment is ICRVBadDebtRepayment {
     function getOraclePrice() public view override returns (uint256) {
         (, int256 price, , , ) = CRV_USD_FEED.latestRoundData();
         if (price <= 0) revert InvalidOracleAnswer();
+        if (price > MAX_ORACLE_PRICE) revert CRVPriceTooHigh(price, MAX_ORACLE_PRICE);
         return uint256(price);
     }
 
     /// @inheritdoc ICRVBadDebtRepayment
     function repay() external returns (uint256) {
-        if (CRV.balanceOf(AaveV2Ethereum.COLLECTOR) == 0) revert NoCRVForRepayment();
-        CRV.approve(address(AaveV2Ethereum.POOL), totalCRVReceived);
-        CRV.safeTransferFrom(AaveV2Ethereum.COLLECTOR, address(this), totalCRVReceived);
-        return AaveV2Ethereum.POOL.repay(address(CRV), totalCRVReceived, 2, BAD_DEBTOR);
+        uint256 balance = CRV.balanceOf(AaveV2Ethereum.COLLECTOR);
```

Comment:
I think there is an edge case, that if the swap rate goes down, and then we are able to repay the debt without spending all the allowance on CRV, and after debt is repaid, this user borrows more CRV and there is still CRV on collector, the user will be able to use this method to keep repaying his new debt until allowance is reached. 
Provably its better to have a check / flag here to see if all debt has been repaid